### PR TITLE
Add-MCP3421-18bit-ADC

### PR DESCRIPTION
**Pull request by tyeth**
Adds the 18bit ADC driver, not tested yet

### CODE REVIEW

**Comment by tyeth** [comment]
Weird, I got an email saying the C3 qtpy failed, but when I clicked on the View Workflow Run it showed only a clang/doxy task failure.

Tested on staging:

**Comment by tyeth** [comment]
@brentru ready for review

File: src/components/i2c/drivers/WipperSnapper_I2C_Driver_MCP3421.h [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    return true;
+  }
+
+  /*******************************************************************************/
+  /*!
+      @brief    Configures the MCP3421 sensor.
+      @returns  True if the sensor was configured successfully, False otherwise.
+  */
+  /*******************************************************************************/
+  bool configureSensor() {
```
Comment (brentru) [inline src/components/i2c/drivers/WipperSnapper_I2C_Driver_MCP3421.h:77]: Love the refactoring outwards like this!

File: src/components/i2c/drivers/WipperSnapper_I2C_Driver_MCP3421.h [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    if (!_mcp3421->startOneShotConversion()) {
+      WS_DEBUG_PRINTLN("Failed to start one-shot conversion");
+      return false;
+    }
+    while (!_mcp3421->isReady()) {
+      ulong newMillis = millis();
+      if (newMillis - start > 500) {
+        WS_DEBUG_PRINTLN("Timeout waiting for conversion to complete");
+        return false;
+      } else if (start > newMillis) {
```
Comment (brentru) [inline src/components/i2c/drivers/WipperSnapper_I2C_Driver_MCP3421.h:120]: I am concerned that the rollover condition may lead to `start` being improperly fed a new value.

You may want to consider removing the `else if ()` and changing the `if` to include the rollover condition:
```
if (newMillis - start >= 500)
```

**Review by brentru — CHANGES_REQUESTED** [review]

**Review by brentru — APPROVED** [review]

**Pull request merged**